protocol/message_validator.go

```diff
@@ -3,9 +3,92 @@ package protocol
 import (
 	"errors"
 	"github.com/status-im/status-go/protocol/protobuf"
+	"strconv"
 	"strings"
 )
 
+func ValidateReceivedSendTransaction(message *protobuf.SendTransaction) error {
```

Comment:
Maybe having something like this would be useful:
```go
type Validator struct {
    message interface{}
}

func NewValidator(message interface{}) *Validator {
    return &Validator{message: message}
}

func (v *Validator) Validate() error {
    switch vm := v.message.(type) {
    case *protobuf.SendTransaction:
        return v.validateSendTransaction()
    case *protobuf.RequestAddressForTransaction
        return ...
    }
}
```

Or even have a single function `func ValidateMessage(message interface{}) error` with this switch inside. This way from all the places you could call `ValidateMessage`.